lib/parser/basic-parsers.h

```diff
@@ -1343,7 +1343,14 @@ template<typename PA> class SourcedParser {
     const char *start{state.GetLocation()};
     auto result{parser_.Parse(state)};
     if (result.has_value()) {
-      result->source = CharBlock{start, state.GetLocation()};
+      const char *end{state.GetLocation()};
+      if (start < end && start[0] == ' ') {
```

Comment:
Maybe these should be `while` loops out of an overabundance of caution?

Reply:
OK. I actually had it that way initially.